ESP8266Interface.cpp

```diff
@@ -381,7 +381,7 @@ int ESP8266Interface::socket_recv(void *handle, void *data, unsigned size)
     int32_t recv;
     if (socket->proto == NSAPI_TCP) {
         recv = _esp.recv_tcp(socket->id, data, size);
-        if (recv <= 0) {
+        if (recv <= 0 && recv != NSAPI_ERROR_WOULD_BLOCK) {
```

Comment:
One question... Is there any situation where these conditions would be met and the socket wouldn't have been closed in `ESP8266.cpp`? If this flag ever gets set to false incorrectly `socket_close()` won't actually try to close the socket when called.